core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/cassandra/NumericDataMapper.java

```diff
@@ -70,28 +67,32 @@ public NumericDataMapper(boolean includeWriteTime) {
 
     @Override
     public List<NumericData> apply(ResultSet resultSet) {
-        if (resultSet.isExhausted()) {
-            return Collections.emptyList();
-        }
-        Row firstRow = resultSet.one();
-        NumericMetric metric = getMetric(firstRow);
-        metric.addData(rowConverter.getData(firstRow));
-
-        for (Row row : resultSet) {
-            metric.addData(rowConverter.getData(row));
-        }
-
-        return metric.getData();
+        List<NumericData> data = new ArrayList<>();
+
+        return StreamSupport.stream(resultSet.spliterator(), false).map(row ->
+            new NumericData(row.getUUID(ColumnIndex.TIME.ordinal()), row.getDouble(ColumnIndex.VALUE.ordinal()),
+                row.getMap(ColumnIndex.TAGS.ordinal(), String.class, String.class))
+        ).collect(toList());
+
+//        Row firstRow = resultSet.one();
+//        NumericMetric metric = getMetric(firstRow);
+//        metric.addData(rowConverter.getData(firstRow));
+//
+//        for (Row row : resultSet) {
+//            metric.addData(rowConverter.getData(row));
+//        }
+//
+//        return metric.getData();
     }
 
-    private NumericMetric getMetric(Row row) {
-        NumericMetric metric = new NumericMetric(row.getString(ColumnIndex.TENANT_ID.ordinal()), getId(row),
-                row.getMap(ColumnIndex.METRIC_TAGS.ordinal(), String.class, String.class),
-                row.getInt(ColumnIndex.DATA_RETENTION.ordinal()));
-        metric.setDpart(row.getLong(ColumnIndex.DPART.ordinal()));
-
-        return metric;
-    }
+//    private NumericMetric getMetric(Row row) {
```

Comment:
No mercy: rename to `rowToMetric` and reuse in `apply`... or remove commented code